### ISSUE TITLE
Mqtt 3 connection without using certificate validation

### ISSUE THREAD

**Issue by BobClaerhout** [eclipse/ditto#679]
We recently installed Ditto version 1.1.0 from version 1.0.0 but we aren't able to communicate with the Azure HUB over mqtt anymore. I'm using the exact same connection as it was before (did an export of the database in the old cluster and creating the connection with the same parameters) and still we cannot connect to that host again. (fyi: we stopped the connection in the old cluster since azure HUB cannot have 2 simultanuous connections with the same credentials).   
To try figure out what's going on. I've created a connection to test.mosquitto.org:1883 which is working fine. From the moment I change the host to ssl://test.mosqtuitto.org:1883 (and verifyCertificates is set to false). I'm getting following error in the logs:
```
2020-05-11 14:02:41,138 WARN  [] c.h.c.i.m.h.d.MqttDisconnectHandler  - Exception while disconnecting: {}
io.netty.handler.codec.DecoderException: java.lang.NullPointerException
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
        at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:792)
        at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:475)
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:831)
Caused by: java.lang.NullPointerException: null
        at java.base/java.util.Objects.requireNonNull(Objects.java:221)
        at java.base/java.util.Arrays$ArrayList.<init>(Arrays.java:4354)
        at java.base/java.util.Arrays.asList(Arrays.java:4341)
        at org.eclipse.ditto.services.connectivity.messaging.internal.ssl.DittoTrustManager.isServerCertificateInTrustStore(DittoTrustManager.java:94)
        at org.eclipse.ditto.services.connectivity.messaging.internal.ssl.DittoTrustManager.checkServerTrusted(DittoTrustManager.java:83)
        at java.base/sun.security.ssl.AbstractTrustManagerWrapper.checkServerTrusted(SSLContextImpl.java:1524)
        at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:619)
        at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.onCertificate(CertificateMessage.java:460)
        at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.consume(CertificateMessage.java:360)
        at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:392)
        at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:443)
        at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1061)
        at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1048)
```
In the stacktrace, you can find following line: `org.eclipse.ditto.services.connectivity.messaging.internal.ssl.DittoTrustManager.isServerCertificateInTrustStore(DittoTrustManager.java:94)`. It's a bit weird to me since I configured it to not check the certificates. Did something change from 1.0.0 to 1.1.0 considering this?  

I noticed following issue: eclipse/ditto#547. This issue is about support for MQTT 5 in ditto so it might be related to this?

**Comment by w4tsn** [comment]
I'm not sure yet, if mqtt 5 is really connected to this. I've not touched mqtt3 to my knowledge, except for little refactoring which left almost everything as it is for mqtt3.

I'll have a look into the changes introduced via #547 and see if I can find something.

> ssl://test.mosqtuitto.org:1883

I suppose the typo is not in your actual configuration ;) Apart from that I don't think it'll work to use an ssl connection to the non-ssl endpoint on 1883. At least it should not as I understand it.

From the documentation I read that 

> validateCertificates: Must be true

Which sounds pretty mandatory to me. Not sure about that.

That said, I don't yet understand why it worked before 1.0.0 and now it doesn't. Can you provide more context?

**Comment by thjaeckle** [comment]
This is clearly a bug in Ditto which was easily spotted and fixed in #688
